tests/attention/test_cudnn_decode.py

```
@@ -170,3 +287,249 @@ def test_cudnn_decode(
+    num_kv_heads, num_qo_heads, head_dim = 2, 4, 128
+
+    q = torch.randn(
+        batch_size, num_qo_heads, head_dim, device=device, dtype=torch.float32
+    )
+    k_cache, v_cache, block_tables = _build_paged_kv(
+        batch_size, s_kv, page_size, num_kv_heads, head_dim, torch.float32, device
+    )
+    actual_seq_lens_kv = torch.full(
+        (batch_size, 1, 1, 1), s_kv, dtype=torch.int32, device=device
+    )
+    with pytest.raises(ValueError, match="only supports torch.float16"):
```

Ruff (0.16.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)